modules/custom/activity_viewer/src/Plugin/views/row/ActivityRow.php

```diff
@@ -30,8 +31,37 @@ public function preRender($result) {
 
       foreach ($result as $row) {
         $render_result = [];
-        $render_result[] = $row;
         $entity = $row->_entity;
+        $target_type = $entity->get('field_activity_entity')->target_type;
+        $render_entity = TRUE;
+
+        if (!empty($target_type) && $target_type === "comment") {
+          $comment_id = $entity->get('field_activity_entity')->target_id;
+          $comment_storage = \Drupal::entityTypeManager()->getStorage('comment');
+          $comment = $comment_storage->load($comment_id);
+          $parent_id = $comment->get('entity_id')->target_id;
+          $parent_type = $comment->get('entity_type')->value;
+
+          if (!empty($parent_id) && !empty($parent_type)) {
+            if ($parent_type === "post") {
+              $parent_storage = \Drupal::entityTypeManager()->getStorage($parent_type);
+              $parent_entity = $parent_storage->load($parent_id);
+              $current_user_id = \Drupal::currentUser()->id();
```

Comment:
I think this comes already with a `->getAccount()` method which will return the current User.. so you dont have to do a separate load for this.